challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map(&:capitalize)
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |x| x == "dog"} #implement your solution here
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |x| x == "dog"}#implement your solution here
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count#implement your solution here
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)#implement your solution here
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.grep(/cat|dog/)#implement your solution here
```

Comment:
While this is fairly legit given the inputs we know about, I suspect we want any entries that start with "c" or "d". So, if "dragon" shows up in the `input` array, this method should pull it out too. I'm pretty sure there's a method on [`String`](https://ruby-doc.org/core-2.5.0/String.html) that does just that.